presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java

```diff
@@ -85,16 +85,79 @@ public String explain(CreateTable statement, List<Expression> parameters)
         checkArgument(!statement.getElements().isEmpty(), "no columns for table");
 
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getName());
+
+        LinkedHashMap<String, ColumnMetadata> columns = new LinkedHashMap<>();
+        Map<String, Object> inheritedProperties = new HashMap<>();
+        prepareColumnMetadata(statement, metadata, session, tableName, columns, inheritedProperties);
+
         Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
         if (tableHandle.isPresent()) {
+            TableMetadata tableMetadata = metadata.getTableMetadata(session, tableHandle.get());
             if (!statement.isNotExists()) {
-                throw new SemanticException(TABLE_ALREADY_EXISTS, statement, "Table '%s' already exists", tableName);
+                if (columnsEquals(columns.values(), tableMetadata.getColumns())) {
```

Comment:
I don't think we should add this check. It changes the semantics of the operation and seems to be arbitrary in how it compares the schema. For example, it ignores properties, which can be a fundamental part of the table definition.